### PR TITLE
[dist, ci] feat: add FSDP2 equivalence integration tests for Qwen3

### PR DESCRIPTION
**Pull request by TimYangst**
### What does this PR do?

> Add a comprehensive FSDP2 integration test suite under `tests/distributed/` that validates VeOmni's FSDP2 distributed training produces numerically equivalent results to a single-GPU HuggingFace baseline. Covers pure FSDP2 and FSDP2+Ulysses sequence parallelism modes using Qwen3-0.6B on 8 GPUs.

### Checklist Before Starting

- [ ] Search for similar PRs. Paste at least one query link here: ...
- [ ] Format the PR title as `[{modules}] {type}: {description}` (This will be checked by the CI)

### Test

- **Baseline generation**: Single-GPU HuggingFace training on Qwen3-0.6B with `flash_attention_2`, padded BSH format, recording per-step loss and gradient norms.
- **FSDP2 test**: 8-GPU `torchrun` with `rmpad_with_pos_ids` data format, comparing loss and grad norms against baseline within configurable relative tolerances (default: loss rtol=0.1%, grad_norm rtol=0.6%).
- **FSDP2+Ulysses test**: 8-GPU with Ulysses SP size=4 (dp_size=2), same tolerance checks.
- Test dataset: 64-sample FineWeb subset pre-tokenized with Qwen3 tokenizer (max_seq_len=4096), bundled as a parquet file.

### Design & Code Changes

- **`tests/distributed/test_fsdp2_equivalence.py`** — Main test module:
  - `run_training_single_gpu_baseline()`: Trains with HF original loss/attention (explicitly unpatches VeOmni monkey-patches).
  - `run_training_fsdp2_distributed()`: Trains with VeOmni FSDP2 + gradient accumulation + `veomni_clip_grad_norm`.
  - Three pytest entries: `test_generate_baseline_via_subprocess`, `test_qwen3_fsdp2_via_subprocess`, `test_qwen3_fsdp2_ulysses_via_subprocess`.
  - Assertion helpers for labels shift correctness (SP-enabled vs disabled) and HF baseline implementation verification.
- **`tests/distributed/test_parameters.py`** — `TestArguments` dataclass and argparse CLI for configuring model, data, parallelism, and tolerance parameters.
- **`tests/distributed/dataset.py`** — Deterministic test datasets (`SimpleDeterministicDataset`, `DistributedDeterministicDataset`) with explicit index computation for reproducible DP partitioning, plus `PaddingCollator` and `MainCollator` integration.
- **`tests/distributed/utils.py`** — Helper functions: VeOmni patch unapply (`unapply_all_veomni_patches`), model backend verification, FSDP2 model/optimizer builder, baseline loss calculator.
- **`tests/distributed/results_io.py`** — JSON-based save/load for baseline results with config compatibility verification.
- **`tests/distributed/data/fineweb_64_qwen3_tokenizer_4096.parquet`** — Bundled test dataset.

### Checklist Before Submitting

> [!IMPORTANT]
> Please check all the following items before requesting a review, otherwise the reviewer might deprioritize this PR for review.

- [ ] Read the [Contribute Guide](https://github.com/ByteDance-Seed/VeOmni/blob/main/CONTRIBUTING.md).
- [ ] Apply [pre-commit checks](https://github.com/ByteDance-Seed/VeOmni/blob/main/CONTRIBUTING.md?plain=1#L22): `make commit && make style && make quality`
- [ ] Add / Update [the documentation](https://github.com/ByteDance-Seed/VeOmni/blob/main/docs).
- [ ] Add unit or end-to-end test(s) to [the CI workflow](https://github.com/ByteDance-Seed/VeOmni/tree/main/.github/workflows) to cover all the code. If not feasible, explain why: ...

### CODE REVIEW

File: tests/distributed/test_fsdp2_equivalence.py [hunk 1]
```diff
@@ -0,0 +1,873 @@
...
+    unapply_all_veomni_patches,
+    verify_model_backend,
+)
+
+
+# =============================================================================
+# Constants
+# =============================================================================
+NPROC_PER_NODE = 8  # Fixed for 8-GPU test environments
+QWEN3_MODEL_PATH = "/mnt/hdfs/models/Qwen3-0.6B"
```
Comment (TimYangst) [inline tests/distributed/test_fsdp2_equivalence.py:47]: Hard-coded weights path, need to be updated.
